### PR TITLE
[bilibili] Support episodes

### PR DESCRIPTION
**Pull request by PeterDing**
Support episodes of bilibili.com.

e.g.

http://bangumi.bilibili.com/anime/v/40062

### CODE REVIEW

File: youtube_dl/extractor/bilibili.py [hunk 1]
```diff
@@ -126,16 +145,23 @@ class BiliBiliIE(InfoExtractor):
...
+        if not _is_episode:
+            params = compat_parse_qs(self._search_regex(
+                [r'EmbedPlayer\([^)]+,\s*"([^"]+)"\)',
+                r'<iframe[^>]+src="https://secure\.bilibili\.com/secure,([^"]+)"'],
+                webpage, 'player parameters'))
+            cid = params['cid'][0]
+        else:
+            url_t = 'http://bangumi.bilibili.com/web_api/get_source'
+            js = self._download_json(url_t, video_id,
+                                     data='episode_id={}'.format(video_id),
```
Comment (dstftw) [inline youtube_dl/extractor/bilibili.py]: `{}` does not work in python 2.6.

**Comment by yan12125** [comment]
```bilibili.py``` undergoes a major rewriting after BiliBili.com switches to HTML5-based video players (#10375). Could you rebase your version to the current master?

File: youtube_dl/extractor/bilibili.py [hunk 2]
```diff
@@ -19,9 +19,20 @@
...
+    'Accept-Encoding': 'gzip, deflate',
+    'Accept-Language': 'zh-CN,zh;q=0.8,en;q=0.6,zh-TW;q=0.4',
+    'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/52.0.2743.82 Safari/537.36',
+    'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
+    'Accept': '*/*',
+    'Cache-Control': 'max-age=0',
+    'X-Requested-With': 'XMLHttpRequest',
+    'Connection': 'keep-alive',
+}
+
```
Comment (yan12125) [inline youtube_dl/extractor/bilibili.py]: Are all these headers necessary?

File: youtube_dl/extractor/bilibili.py [hunk 3]
```diff
@@ -118,6 +129,14 @@ class BiliBiliIE(InfoExtractor):
             'skip_download': True,
         },
         'expected_warnings': ['upload time'],
+    }, {
+        'url': 'http://bangumi.bilibili.com/anime/v/40068',
+        'info_dict': {
+            'id': '2533616',
+            'title': '混沌武士 : 第7集 四面楚歌 A Risky Racket',
+            'description': "故事发生在日本的江户时代。风是一个小酒馆的打工女。一日，酒馆里来了一群恶霸，虽然他们的举动令风十分不满，但是毕竟风只是一届女流，无法对他们采取什么行动，只能在心里嘟哝。这时，酒家里又进来了个“不良份子”无幻，说以50个丸子帮她搞定这群人，风觉得他莫名其妙，也就没多搭理他。而在这时，风因为一个意外而将茶水泼在了恶霸头领——龙次郎身上。愤怒的恶霸们欲将风的手指砍掉，风在无奈中大喊道：“丸子100个！”…… 　　另一方面，龙次郎的父亲也就是当地的代官，依仗自己有着雄厚的保镖实力，在当地欺压穷人，当看到一穷人无法交齐足够的钱过桥时，欲下令将其杀死，武士仁看不惯这一幕，于是走上前，与代官的保镖交手了…… 　　酒馆内，因为风答应给无幻100个团子，无幻将恶霸们打败了，就在这时，仁进来了。好战的无幻立刻向仁发了战书，最后两败俱伤，被代官抓入牢房，预计第二天斩首…… 　　得知该状况的风，为报救命之恩，来到了刑场，利用烟花救出了无幻和仁。而风则以救命恩人的身份，命令二人和她一起去寻找带着向日葵香味的武士……(by百科)",
```
Comment (yan12125) [inline youtube_dl/extractor/bilibili.py]: Use the ```'md5:xxx'``` syntax if the value is too long.

Comment (PeterDing) [inline youtube_dl/extractor/bilibili.py]: `DEVELOPER INSTRUCTIONS` writed:
```'md5': 'TODO: md5 sum of the first 10241 bytes of the video file (use --test)',```

Beause `TODO`, I'm confused.
so, md5sum is for total the video file video or for the first 10241 bytes of the video file.

Comment (yan12125) [inline youtube_dl/extractor/bilibili.py]: Something like: ```'description': 'md5:6a9622b911565794c11f25f81d6a97d2'```

**Comment by PeterDing** [comment]
At the last commit, I rebase to the the current master.

File: youtube_dl/extractor/bilibili.py [hunk 4]
```diff
@@ -18,6 +18,7 @@
     'Content-Type': 'application/x-www-form-urlencoded; charset=UTF-8',
     'Accept': '*/*',
     'X-Requested-With': 'XMLHttpRequest',
+    'Referer': 'http://www.bilibili.com/',
 }
```
Comment (PeterDing) [inline youtube_dl/extractor/bilibili.py]: this HEADER is necessary

Comment (yan12125) [inline youtube_dl/extractor/bilibili.py]: I meant other headers. like Accept* ones. Usually they are not necessary.

**Comment by yan12125** [comment]
Please run tests following step 6 of https://github.com/rg3/youtube-dl/blob/master/README.md#adding-support-for-a-new-site

**Comment by yan12125** [comment]
Don't remove other _TESTS. Run ```TestDownload.test_BiliBili_4``` for the 5th test.

**Comment by yan12125** [comment]
Looks good now. Could you squash all commits into a single one?

**Comment by yan12125** [comment]
Thanks!

**Pull request merged**